Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl

```diff
@@ -56,210 +45,289 @@ AABB CombineAABB(AABB aabb0, AABB aabb1)
     return parentAABB;
 }
 
-bool IsLeaf(uint nodeIndex)
-{
-    const uint NumberOfInternalNodes = GetNumInternalNodes(Constants.NumberOfElements);
-    return nodeIndex >= NumberOfInternalNodes;
-}
-
 #define BIT(x) (1 << (x))
 
-[numthreads(THREAD_GROUP_1D_WIDTH, 1, 1)]
-void main( uint3 DTid : SV_DispatchThreadID )
-{
-    if (DTid.x >= Constants.NumberOfElements) return;
+// Must be at least FullTreeletSize
+#define NumThreadsInGroup 32
```

Comment:
This will be interesting to play with on AMD, which would technically want a minimum of 64 threads, in the cases where there's 35 partitions, a higher value would likely be a win, but a loss on NV